app/scripts/modules/core/pipeline/config/stages/createLoadBalancer/createLoadBalancerStage.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular.module('spinnaker.core.pipeline.stage.createLoadBalancerStage', [
+  require('../../../../utils/lodash.js'),
+  require('../../../../cloudProvider/cloudProvider.registry.js'),
+])
+  .config(function(pipelineConfigProvider, settings) {
+
+    // Register this stage only if infrastructure stages are enabled in settings.js
+    if (settings.feature.infrastructureStages) {
+      pipelineConfigProvider.registerStage({
+        key: 'upsertLoadBalancers',
+        label: 'Create Load Balancers',
+        description: 'Creates one or more load balancers',
+        templateUrl: require('./createLoadBalancerStage.html'),
+        executionDetailsUrl: require('./createLoadBalancerExecutionDetails.html'),
+        defaultTimeoutMs: 5 * 60 * 1000, // 5 minutes
+        validators: [
+        ],
+      });
+    }
+
+  })
+  .controller('createLoadBalancerStageCtrl', function($injector, $scope, $uibModal, providerSelectionService,
+                                                         cloudProviderRegistry) {
+
+    function initializeCommand() {
+      $scope.stage.loadBalancers = $scope.stage.loadBalancers || [];
+    }
+
+    this.addLoadBalancer = function() {
+      providerSelectionService.selectProvider($scope.application).then(function(selectedProvider) {
```

Comment:
You'll want to include the feature in the `selectProvider` call, i.e.
```javascript
providerSelectionService.selectProvider($scope.application, 'loadBalancers')
```
Otherwise, you'll get all providers that are configured, not just the ones that support load balancers.